crates/bevy_math/src/direction.rs

```diff
@@ -605,6 +709,29 @@ mod tests {
         assert_eq!(Dir2::new_and_length(Vec2::X * 6.5), Ok((Dir2::X, 6.5)));
     }
 
+    #[test]
+    fn dir2_slerp() {
+        assert!(Dir2::X.slerp(Dir2::Y, 0.5).relative_eq(
+            &Dir2::from_xy(0.5_f32.sqrt(), 0.5_f32.sqrt()).unwrap(),
+            f32::EPSILON,
+            f32::EPSILON
+        ));
+        assert_eq!(Dir2::X.slerp(Dir2::Y, 0.0), Dir2::X);
+        assert!(Dir2::X
+            .slerp(Dir2::Y, 1.0)
+            .relative_eq(&Dir2::Y, f32::EPSILON, f32::EPSILON));
+        assert!(Dir2::X.slerp(Dir2::Y, 1.0 / 3.0).relative_eq(
+            &Dir2::from_xy(0.75_f32.sqrt(), 0.5).unwrap(),
+            f32::EPSILON,
+            f32::EPSILON
+        ));
+        assert!(Dir2::X.slerp(Dir2::Y, 2.0 / 3.0).relative_eq(
+            &Dir2::from_xy(0.5, 0.75_f32.sqrt()).unwrap(),
+            f32::EPSILON,
+            f32::EPSILON
+        ));
```

Comment:
I think it would be good to test clockwise rotations, since the implementation depends on `Vec2::angle_between`.